src/built_ins.md

```diff
@@ -18,222 +18,757 @@ The built-in types for `alan` are:
+```
 
 ##### Logical and Bitwise
 
-* `and(int8, int8): int8`, `and(int16, int16): int16`, `and(int32, int32): int32`, `and(int64, int64): int64`, `and(bool, bool): bool`
-  Applies the `and` truth table to the inputs. If integers, it applies bitwise, if boolean, to the single bit.
-* `or(int8, int8): int8`, `or(int16, int16): int16`, `or(int32, int32): int32`, `or(int64, int64): int64`, `or(bool, bool): bool`
-  Applies the `or` truth table to the inputs. If integers, it applies bitwise, if boolean, to the single bit.
-* `xor(int8, int8): int8`, `xor(int16, int16): int16`, `xor(int32, int32): int32`, `xor(int64, int64): int64`, `xor(bool, bool): bool`
-  Applies the `xor` truth table to the inputs. If integers, it applies bitwise, if boolean, to the single bit.
-* `not(int8): int8`, `not(int16): int16`, `not(int32): int32`, `not(int64): int64`, `not(bool): bool`
-  Applies the `not` truth table to the input. If integers, it applies bitwise, if boolean, to the single bit.
-* `nand(int8, int8): int8`, `nand(int16, int16): int16`, `nand(int32, int32): int32`, `nand(int64, int64): int64`, `nand(bool, bool): bool`
-  Applies the `nand` truth table to the inputs. If integers, it applies bitwise, if boolean, to the single bit.
-* `nor(int8, int8): int8`, `nor(int16, int16): int16`, `nor(int32, int32): int32`, `nor(int64, int64): int64`, `nor(bool, bool): bool`
-  Applies the `nor` truth table to the inputs. If integers, it applies bitwise, if boolean, to the single bit.
-* `xnor(int8, int8): int8`, `xnor(int16, int16): int16`, `xnor(int32, int32): int32`, `xnor(int64, int64): int64`, `xnor(bool, bool): bool`
-  Applies the `xnor` truth table to the inputs. If integers, it applies bitwise, if boolean, to the single bit.
+```rust
+// Return the logical or bitwise `and`
+and(int8, int8): int8
+and(int16, int16): int16
+and(int32, int32): int32
+and(int64, int64): int64
+and(bool, bool): bool
+```
+
+```rust
+// Return the logical or bitwise `or`
+or(int8, int8): int8
+or(int16, int16): int16
+or(int32, int32): int32
+or(int64, int64): int64
+or(bool, bool): bool
+```
+
+```rust
+// Return the logical or bitwise `xor`
+xor(int8, int8): int8
+xor(int16, int16): int16
+xor(int32, int32): int32
+xor(int64, int64): int64
+xor(bool, bool): bool
+```
+
+```rust
+// Return the logical or bitwise `not`
+not(int8): int8
+not(int16): int16
+not(int32): int32
+not(int64): int64
+not(bool): bool
+```
+
+```rust
+// Return the logical or bitwise `nand`
+nand(int8, int8): int8
+nand(int16, int16): int16
+nand(int32, int32): int32
+nand(int64, int64): int64
+nand(bool, bool): bool
+```
+
+```rust
+// Return the logical or bitwise `nor`
+nor(int8, int8): int8
+nor(int16, int16): int16
+nor(int32, int32): int32
+nor(int64, int64): int64
+nor(bool, bool): bool
+```
+
+```rust
+// Return the logical or bitwise `xnor`
+xnor(int8, int8): int8
+xnor(int16, int16): int16
+xnor(int32, int32): int32
+xnor(int64, int64): int64
+xnor(bool, bool): bool
+```
 
 ##### Comparators
```

Comment:
most of these are defined as operators and it might leave people wondering, "do I have to use `gt` instead of `>`?". perhaps let's mention that briefly or move operators further up?

Reply:
I'm not sure how to move them up, since they reference these function names in the table. Perhaps this concern should be included in a blog post / tutorial?

Reply:
I see. Let's put it right after then? I don't think this needs a blog post

Reply:
Isn't it right after right now?

Reply:
ah got disoriented within the docs, nvm :)